src/input/touch.c

```diff
@@ -214,9 +299,30 @@ int touch_to_mouse(void)
             mouse_reset_scroll();
             mouse_reset_button_state();
             return 1;
+        } else if (data.touchpad_mode_click_type != EMULATED_MOUSE_CLICK_NONE) {
+            return handle_emulated_mouse_clicks();
         }
         return 0;
     }
-    mouse_set_from_touch(first, get_latest_touch());
+    switch (data.mode) {
+        case TOUCH_MODE_TOUCHPAD:
+            handle_mouse_touchpad();
+            break;
+        case TOUCH_MODE_DIRECT:
+            handle_mouse_direct();
+            break;
+        default:
+            mouse_set_from_touch(first, get_latest_touch());
```

Comment:
Also add a `break;` here:
```suggestion
            mouse_set_from_touch(first, get_latest_touch());
            break;
```